File: src/output/format.ts
import { outputJson } from './json.js';
import { outputTable } from './table.js';

export function output(data: unknown, format: string): void {
  if (format === 'table' && Array.isArray(data)) {
    outputTable(data as Record<string, unknown>[]);
  } else {
    outputJson(data);
  }
}

File: src/tableau_cli/output/format.py
from __future__ import annotations

from typing import Any

from .json_output import output_json
from .table_output import output_table


def output(data: Any, fmt: str) -> None:
    if fmt == "table" and isinstance(data, list):
        output_table(data)
    else:
        output_json(data)
